msg/service.go

```diff
@@ -76,6 +76,10 @@ func PathWithWildcard(s string) (string, bool) {
 		if k == "*" {
 			return path.Join(append([]string{"/skydns/"}, l[:i]...)...), true
 		}
+		if strings.HasPrefix(k, "_") {
```

Comment:
Ah, I added a range-loop here myself.. Ah. Still: move your stuff in the loop above :)